lib/portage/dbapi/porttree.py

```diff
@@ -645,6 +646,12 @@ def _pull_valid_cache(self, cpv, ebuild_path, repo_path):
                 # EAPI from _parse_eapi_ebuild_head, we disregard cache entries
                 # for unsupported EAPIs.
                 continue
+
+            # egencache for example needs to be able to invalidate and force
+            # fresh entries.
+            if not force and not repo.volatile:
+                break
```

Comment:
It's not really clear which auxdb we've trusted here. Maybe we should only trust pregen_auxdb since that's the one that's distributed with the repository. However, it's expected for pregen_auxdb to contain stale metadata if the repository inherits eclasses from another repository that have changed.